src/theory/bags/solver_state.h

```diff
@@ -84,7 +90,7 @@ class SolverState : public TheoryState
   /** collection of bag representatives */
   std::set<Node> d_bags;
   /** bag -> associated elements */
-  std::map<Node, std::set<Node>> d_bagElements;
+  std::map<Node, std::vector<std::pair<Node, Node>>> d_bagElements;
```

Comment:
Can you document the pair here?

Reply:
Also, this seems to be a fairly important data structure, can you document a bit more the lifetime of this?  Is this cleared on full effort? When is it valid?